_gsocblogs/2024/blog_ATLCompression_IshanDarji.md

```diff
@@ -0,0 +1,120 @@
+
+In the end, to compare each library with the most similar compression speed to zlib at compression level 1 we see:
+| Library | Level | Ratio | Compression | Decompression |
+|---------|-------|-------|-------------|---------------|
+|zlib | 1 | 62.23 | 67.7 MB/s |  234 MB/s |
+|zstd | 8 | 58.35 | 65.6 MB/s |  525 MB/s |
+|brotli | 4 | 57.74 | 57.1 MB/s |  268 MB/s| 
+
+Whereas when looking at the most similar compression ratios  we see:
+| Library | Level | Ratio | Compression | Decompression |
+|---------|-------|-------|-------------|---------------|
+|zlib | 1 | 62.23 | 67.7 MB/s |  234 MB/s |
+|zstd | 1 | 62.61 | 457 MB/s  | 670 MB/s |
+|brotli | 1 | 62.79 | 233 MB/s  | 225 MB/s |
+
+*lzlib and xz are not exactly comparable, their compression is much slower but with much lower ratios across their entire operable range.
+
+Here we see that if we are looking to get a similar compression ratio, zlib performs the slowest by far at close to a fourth of the speed brotli, and zstd reaches almost two times brotli's speed. If we are looking to get similar compression times, while we do not have as close of a match as compression ratio, both zstd and brotli result in a reduction of a few percent of the original filesize.  
+
+Given the data collected, we chose to use zstd as our compression algorithm. This decision came for many reasons and was primarily determined by the analysis above. The goal for this project was to find a faster replacement for the currently in-place Zlib compression system, with particular importance on compression ratio rather than compression speed. The reason for this is that when we look at the scale of data generated by the ATLAS experiment, a reduction of even a few percentages makes a large difference when the filesizes are upwards of exabytes. The two most relevant graphs for this comparison are the two smaller ones looking at zlib, zstd, and brotli at fixed metrics. For example, we see that if we wanted to keep the compression ratio of the system the same, switching from zlib to zstd would still yield improvements of roughly 7 times in compression speed, and 3 times in decompression speed. However, because we are looking to keep the compute resources required roughly the same, we instead hold the compression speed as close to fixed as possible. This gives us the first of the two smaller tables above where we see that, in terms of compression speed, zlib at level 1, zstd at level 8, and brotli at level 4 are very roughly equal. We now see slight differences in the compression ratio, where brotli offers the best compression, followed closely by zstd, and zlib trails behind by a slightly more notable portion. 
```

Comment:
I would add a statement that this study suggests improvements of ~6% in compression ratio at the same CPU cost compared to current method. You may want to add that is a significant as even modest gains can translate into massive savings in storage costs at exabyte scale.